Sources/Data Model/Audience/UserAttribute.swift

```diff
@@ -90,49 +90,55 @@ extension UserAttribute {
     
     func evaluate(attributes: OptimizelyAttributes?) throws -> Bool {
         
+        let conditionString = Utils.getConditionString(conditions: self)
```

Comment:
This conditionString buidling is expensive. Let's remove this from normal flow (non-error).
We can build it in each error case.